ide/coqide/coqide_ui.ml

```diff
@@ -98,9 +99,11 @@ let init () =
 \n    <menuitem action='Forward' />\
 \n    <menuitem action='Backward' />\
 \n    <menuitem action='Run to cursor' />\
+\n    <menuitem action='Reset' />\
 \n    <menuitem action='Run to end' />\
+\n    <menuitem action='Force' />\
 \n    <menuitem action='Interrupt' />\
-\n    <menuitem action='Reset' />\
+\n    <separator/>\
 \n    <menuitem action='Previous' />\
 \n    <menuitem action='Next' />\
```

Comment:
FWIW If you read the code, this searches for the next/prev instance of the current word in the buffer.  This should be integrated into the general text search; use CTRL F for next and SHIFT CTRL F for previous.

Reply:
Yes, I’ve seen this feature described somewhere (in a bug report, maybe). So I added separators… I don’t know in which case this feature would be so useful; as it was put in the toolbar, I tend to think that someone finds it very, very useful. But maybe the procedure was just to put all Navigation menu items in the toolbar.

Shift-Ctrl-F is usually used for “Find in Files” (or “Find in all opened tabs”). Ctrl-F is used by the common “Find / Replace.” So I don’t understand your idea. Do you want these menu items to be put in the Edit menu (under the “Find Previous” menu item)? I think it’s correct to have them in the Navigation menu.

Reply:
I didn't explain in full.  I think searching should work more like #14040.  If the search bar isn't shown, Ctrl-F shows it.  If the search bar is shown, Ctrl-F means to go to the next item.  It could also be Ctrl-F3 and F3.  I don't know when or if I will get to implement that, though.

> Shift-Ctrl-F is usually used for “Find in Files” (or “Find in all opened tabs”).

A very subjective statement.

Reply:
> If the search bar is shown, Ctrl-F means to go to the next item. It could also be Ctrl-F3 and F3.

Currently, it is F3 and Shift-F3, which is standard (it works in Firefox for instance) and works whether or not the dialog is shown. So this is in place.

> A very subjective statement.

This is also my manner to say that I agree with this convention I’ve seen used in at least a couple of apps.

> I don't know when or if I will get to implement that, though.

I can do it if you want, though I want to say that the current dialog is more my style, because I like the very explicit text labels. I don’t think your idea will improve the productivity of people. But if it’s important for some important CoqIDE users, it’s still useful.